go/mysql/replication_status.go

```diff
@@ -179,6 +179,11 @@ func ProtoToReplicationStatus(s *replicationdatapb.Status) ReplicationStatus {
 // provided as a list of ReplicationStatus's. This method only works if the flavor for all retrieved ReplicationStatus's is MySQL.
 // The result is returned as a Mysql56GTIDSet, each of whose elements is a found errant GTID.
 func (s *ReplicationStatus) FindErrantGTIDs(otherReplicaStatuses []*ReplicationStatus) (Mysql56GTIDSet, error) {
+	if len(otherReplicaStatuses) == 0 {
+		// If there is nothing to compare this replica against, then we must assume that its GTID set is the correct one.
+		return nil, nil
+	}
```

Comment:
I think that we should note/comment that this function:
  - Is best effort (in this new case we can't know)
  - Can return false positives: not all replicas' IO threads will be in sync and `s` may simply be further ahead and have a larger `Retrieved_Gtid_Set`
      - This is true even with semi-sync as we only require 1 ACKer by default AFAIK 
  - Thus should only be used when the old primary is no longer available (ERS) as comparing the replica with the old primary is the only sure way to check for errant GTIDs

Make sense, or am I missing something(s)?

Reply:
yeah same comment, this can result in data loss .. are we ok with that..

Reply:
I'll add a comment but there won't be false positives or rather there shouldn't be. The function doesn't mark the GTIDs as errant if they come from the current primary. It can miss some GTIDs which are errant, but not the other way around.

Reply:
Added a comment.